src/main/java/core/basesyntax/strategy/handler/impl/BalanceHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.handler.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction.getQuantity() >= 0) {

```

Comment:
you can invert the if, firstly check:
```suggestion
        if (transaction.getQuantity() < 0) {
```
throw an exception
and get rid of `else` block.